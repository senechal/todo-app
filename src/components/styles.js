import styled from 'styled-components';
import Delete from 'assets/delete.svg';

export const Container = styled.div`
    border: 1px solid #d6d6d6;
    margin: 0 auto;
    width: 300px;
    display: flex;
    justify-content: flex-start;
    flex-direction: column;
`;

export const ListHeader = styled.div`
    height: 56px;
    border-bottom: 1px solid #e6e6e6;
    display: flex;
    align-items: center;
    padding: 0 8px;
    color: #e6e6e6;
    background-color: #fefefe;
`;

export const ListFooter = styled.div`
    margin-top: auto;
    height: 32px;
    border-top: 1px solid #e6e6e6;
    display: flex;
    align-items: center;
    padding: 0 8px 0px 12px;
    color: #4d4d4d;
`;

export const List = styled.div`
    list-style-type: none;
    margin: 0;
    padding: 0;
    min-height: 120px;
    max-height: 360px;
    overflow: auto;
`;

export const Item = styled.div`
    color: #4d4d4d;
    display: flex;
    align-items: center;
    padding: 0 8px;
    height: 40px;
    border-bottom: 1px solid #e6e6e6;
    display:flex;
    align-items: center;
    box-sizing: border-box;

    &:last-child:not(:nth-child(-n+2)){
        border-bottom: 0px;
    }
`;

export const Checkbox = styled.input.attrs({ type: 'checkbox' })`
   margin-right: 8px;
`;

export const AddButton = styled.button`
    background-color: #fff;
    width: 24px;
    height: 24px;
    padding: 2.4px 0;
    font-size: 14px;
    vertical-align: -1px;
    border-radius: 4px;
    border: 1px solid #1890ff;
    cursor: pointer;
    margin-left: auto;
    background-color: #1890ff;
    color: #fff;

    &:disabled{
        opacity: 0.5;
        cursor: not-allowed;
    }
`;

export const DeleteButton = styled(AddButton)`
    background-color: #fff;
    border-color: #e6e6e6;
    color: #e6e6e6;

    & svg {
        fill: #e6e6e6;
    }
`;

export const ItemContent = styled.div`
    color: #4d4d4d
    font-size: 16px;
    height: 20px;
    box-sizing: border-box;
    display: ${({ hide }) => (hide ? 'none' : null)};
    width: calc(100% - 62px);
    &:hover{
        border-bottom: 1px solid  #e6e6e6;
    }
`;

export const Input = styled.input.attrs({ type: 'text' })`
    color:#4d4d4d;
    outline: 0;
    border-width: 0 0 1px;
    border-color: #4d4d4d;
    font-size: 16px;
    height: 20px;
    box-sizing: border-box;
    display: ${({ hide }) => (hide ? 'none' : null)};
    width: calc(100% - 62px);
`;

export const DeleteIcon = styled(Delete)`
    max-width: 16px;
    max-height: 16px;
`;
