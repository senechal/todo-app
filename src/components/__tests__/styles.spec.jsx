import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import * as styles from '../styles';

describe('Testing renders for style components', () => {
  const { ItemContent, Input, ...uncontrolledComponents } = styles;

  describe('Uncontrolled components', () => {
    Object.keys(uncontrolledComponents).forEach((displayName) => {
      describe(displayName, () => {
        it('should render correctly', () => {
          const Component = uncontrolledComponents[displayName];
          const { container } = render(<Component />);
          expect(container).toMatchSnapshot();
        });
      });
    });
  });
  describe('Controlled components', () => {
    describe('ItemContent', () => {
      it('should render correctly with hide = false', () => {
        const { container } = render(<ItemContent hide={false} />);
        expect(container).toMatchSnapshot();
      });

      it('should render correctly with hide = true', () => {
        const { container } = render(<ItemContent hide />);
        expect(container).toMatchSnapshot();
      });
    });
    describe('Input', () => {
      it('should render correctly with hide = false', () => {
        const { container } = render(<Input hide={false} />);
        expect(container).toMatchSnapshot();
      });

      it('should render correctly with hide = true', () => {
        const { container } = render(<Input hide />);
        expect(container).toMatchSnapshot();
      });
    });
  });
});
