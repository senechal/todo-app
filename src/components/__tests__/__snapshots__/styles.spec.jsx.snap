// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing renders for style components Controlled components Input should render correctly with hide = false 1`] = `
.c0 {
  color: #4d4d4d;
  outline: 0;
  border-width: 0 0 1px;
  border-color: #4d4d4d;
  font-size: 16px;
  height: 20px;
  box-sizing: border-box;
  width: calc(100% - 62px);
}

<div>
  <input
    class="c0"
    type="text"
  />
</div>
`;

exports[`Testing renders for style components Controlled components Input should render correctly with hide = true 1`] = `
.c0 {
  color: #4d4d4d;
  outline: 0;
  border-width: 0 0 1px;
  border-color: #4d4d4d;
  font-size: 16px;
  height: 20px;
  box-sizing: border-box;
  display: none;
  width: calc(100% - 62px);
}

<div>
  <input
    class="c0"
    type="text"
  />
</div>
`;

exports[`Testing renders for style components Controlled components ItemContent should render correctly with hide = false 1`] = `
.c0 {
  color: #4d4d4d font-size:16px;
  height: 20px;
  box-sizing: border-box;
  width: calc(100% - 62px);
}

.c0:hover {
  border-bottom: 1px solid #e6e6e6;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Testing renders for style components Controlled components ItemContent should render correctly with hide = true 1`] = `
.c0 {
  color: #4d4d4d font-size:16px;
  height: 20px;
  box-sizing: border-box;
  display: none;
  width: calc(100% - 62px);
}

.c0:hover {
  border-bottom: 1px solid #e6e6e6;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Testing renders for style components Uncontrolled components AddButton should render correctly 1`] = `
.c0 {
  background-color: #fff;
  width: 24px;
  height: 24px;
  padding: 2.4px 0;
  font-size: 14px;
  vertical-align: -1px;
  border-radius: 4px;
  border: 1px solid #1890ff;
  cursor: pointer;
  margin-left: auto;
  background-color: #1890ff;
  color: #fff;
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`Testing renders for style components Uncontrolled components Checkbox should render correctly 1`] = `
.c0 {
  margin-right: 8px;
}

<div>
  <input
    class="c0"
    type="checkbox"
  />
</div>
`;

exports[`Testing renders for style components Uncontrolled components Container should render correctly 1`] = `
.c0 {
  border: 1px solid #d6d6d6;
  margin: 0 auto;
  width: 300px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Testing renders for style components Uncontrolled components DeleteButton should render correctly 1`] = `
.c0 {
  background-color: #fff;
  width: 24px;
  height: 24px;
  padding: 2.4px 0;
  font-size: 14px;
  vertical-align: -1px;
  border-radius: 4px;
  border: 1px solid #1890ff;
  cursor: pointer;
  margin-left: auto;
  background-color: #1890ff;
  color: #fff;
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1 {
  background-color: #fff;
  border-color: #e6e6e6;
  color: #e6e6e6;
}

.c1 svg {
  fill: #e6e6e6;
}

<div>
  <button
    class="c0 c1"
  />
</div>
`;

exports[`Testing renders for style components Uncontrolled components DeleteIcon should render correctly 1`] = `
.c0 {
  max-width: 16px;
  max-height: 16px;
}

<div>
  <svg
    class="c0"
    data-file-name="SvgDelete"
  />
</div>
`;

exports[`Testing renders for style components Uncontrolled components Item should render correctly 1`] = `
.c0 {
  color: #4d4d4d;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 8px;
  height: 40px;
  border-bottom: 1px solid #e6e6e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
}

.c0:last-child:not(:nth-child(-n+2)) {
  border-bottom: 0px;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Testing renders for style components Uncontrolled components List should render correctly 1`] = `
.c0 {
  list-style-type: none;
  margin: 0;
  padding: 0;
  min-height: 120px;
  max-height: 360px;
  overflow: auto;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Testing renders for style components Uncontrolled components ListFooter should render correctly 1`] = `
.c0 {
  margin-top: auto;
  height: 32px;
  border-top: 1px solid #e6e6e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 8px 0px 12px;
  color: #4d4d4d;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Testing renders for style components Uncontrolled components ListHeader should render correctly 1`] = `
.c0 {
  height: 56px;
  border-bottom: 1px solid #e6e6e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 8px;
  color: #e6e6e6;
  background-color: #fefefe;
}

<div>
  <div
    class="c0"
  />
</div>
`;
